src/cat.jl

```diff
@@ -24,23 +23,25 @@ struct ConcatDiskArray{T,N,P,C,HC,ID} <: AbstractDiskArray{T,N}
     innerdims::Val{ID}
 end
 
-function ConcatDiskArray(arrays::AbstractArray{Union{<:AbstractArray,Missing}})
+function ConcatDiskArray(arrays::AbstractArray{Union{<:AbstractArray,Missing}}; fill=missing)
     et = Base.nonmissingtype(eltype(arrays))
-    T = Union{Missing,eltype(et)}
+    T = promotetype(typeof(fill), eltype(et))
```

Comment:
```suggestion
    T = Union{Missing,eltype(et)}
```

Reply:
Don't you have to read all the values and promote them? Missing is not necessarily in the type?